src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -460,7 +767,7 @@ def compare_datasets_by_data_type(
     # check that they have columns in common to compare
     elif not has_common_cols(staging_dataset, main_dataset):
```

Comment:
Right, I forgot why I separated them in the first place (maybe I'll remember when I work on this). I can build them into the same function.

Reply:
oh i separated them because `has_common_cols` needs both datasets as input while the `is_valid_dataset` is designed for individual datasets/the namespace they have. I think I'll keep them as is just because that would over complicate that one function (more parameters). I wonder if instead of messages in the comparison report, we should just have them throw exceptions.

Reply:
re Exceptions: Does failing any one of these checks indicate that there's something seriously wrong with the Parquet data? 
* No columns in one of the datasets (What if the dataset has no rows?)
* Duplicate columns
* No intersecting columns

If so, I think an exception makes sense. But if it's something that could theoretically happen but doesn't indicate any _wrongness_ or other error in the production of the Parquet data then it's better to have the comparison report.

Reply:
> re Exceptions: Does failing any one of these checks indicate that there's something seriously wrong with the Parquet data?

    No columns in one of the datasets (What if the dataset has no rows?)
    Duplicate columns
    No intersecting columns

Yeah I think `get_parquet_dataset` won't throw an exception if the dataset is read in empty or has empty cols (as long as the filter values is `None` as well) Agree those three situations should be exceptions as they shouldn't happen to the data. I'll convert to exceptions.